save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/OrganizationController.kt

```diff
@@ -341,20 +342,32 @@ internal class OrganizationController(
             "Could not find an organization with name $organizationName."
         }
         .filter {
-            organizationPermissionEvaluator.hasPermission(authentication, it, Permission.DELETE)
+            organizationPermissionEvaluator.hasPermissionToChangeStatus(authentication, it, status)
         }
         .switchIfEmptyToResponseException(HttpStatus.FORBIDDEN) {
-            "Not enough permission for deletion of organization $organizationName."
+            "Not enough permission for this action with organization of organization $organizationName."
         }
         .filter {
-            !organizationService.hasProjects(organizationName)
+            status == OrganizationStatus.DELETED && !organizationService.hasProjects(organizationName)
         }
         .switchIfEmptyToResponseException(HttpStatus.CONFLICT) {
             "There are projects connected to $organizationName. Please delete all of them and try again."
         }
-        .map {
-            organizationService.deleteOrganization(it.name)
-            ResponseEntity.ok("Organization deleted")
+        .map {organization ->
+            when (status) {
+                OrganizationStatus.BANNED -> {
+                    organizationService.banOrganization(organization)
+                    ResponseEntity.ok("Organization banned")
+                }
+                OrganizationStatus.DELETED -> {
+                    organizationService.deleteOrganization(organization)
+                    ResponseEntity.ok("Organization deleted")
+                }
+                OrganizationStatus.CREATED -> {
+                    organizationService.recoverOrganization(organization)
+                    ResponseEntity.ok("Organization recovered")
```

Comment:
Please do something like `Successfully ***ACTION*** the organization.`

Reply:
done